test/serverExplorer.test.ts

```diff
@@ -0,0 +1,193 @@
+        expect(serviceStub).calledOnce;
+        expect(clearStub).calledOnce;
+        expect(clearStub).calledOnceWith(updateserver);
+    });
+
+    test('serverExplorer.showOutput call should show servers output channel', () => {
+        const showStub = sandbox.stub(serverExplorer.serverOutputChannels, 'get').returns({
+            show: () => {}
+        });
+        serverExplorer.showOutput(serverHandle);
+        expect(showStub).calledOnce;
+
+    });
+
+    test('serverExplorer.getTreeItem call should return TreeItem instance with corresponding label', () => {
+        const serverHandleStub = sandbox.stub(serverExplorer.serverStatus, 'get').resolves({
+            serverHandle
+        });
+        serverExplorer.getTreeItem(serverHandle);
+        expect(serverHandleStub).calledOnce;
+    });
+
+    test('should able to getChildren of the server', () => {
+        const clearStub = sandbox.stub(serverExplorer.servers, 'values').returns([]);
+        const getChildren = serverExplorer.getChildren(undefined);
+        expect(getChildren).deep.equals(Array.from(clearStub));
+    });
+
+    test('serverExplorer.addServerOutput call should show ServerOutput', () => {
+        const getStub = sandbox.stub(serverExplorer.serverOutputChannels, 'get').returns(undefined);
+        serverExplorer.addServerOutput(ProcessOutput);
+        expect(getStub).calledOnce;
+    });
+
+    test('serverExplorer.addServerOutput call should show ServerOutput', () => {
+        const getStub = sandbox.stub(serverExplorer.serverOutputChannels, 'get').returns({
+            show: () => {},
+            append: () => {}
+        });
+        serverExplorer.addServerOutput(ProcessOutput);
+        expect(getStub).calledOnce;
+    });
+
+    test('serverExplorer.refresh call should able check server refresh', () => {
+        const fireStub = sandbox.stub(EventEmitter.prototype, 'fire');
+        serverExplorer.refresh(serverHandle);
+        expect(fireStub).calledOnce;
+        expect(fireStub).calledOnceWith(serverHandle);
+    });
+
+    test('serverExplorer.addLocation should ask for location of the server and name if server detected in provided location', async () => {
+        const showOpenDialogStub = sinon.stub(window, 'showOpenDialog').resolves([{fsPath: 'path/path'}]);
+        await serverExplorer.addLocation();
+        expect(showOpenDialogStub).calledOnce;
+    });
+
+    test('serverExplorer.addLocation should call client.createServerAsync with detected server bean for location and name provided by user', async () => {
+        const findServerStub = sandbox.stub(clientStub, 'findServerBeans').resolves([findServerBeans]);
+        sandbox.stub(window, 'showInputBox').resolves('eap');
+        sandbox.stub(clientStub, 'createServerAsync').resolves(status);
+        await serverExplorer.addLocation();
+        expect(findServerStub).calledOnce;
+    });
+});
+
+suite('serverExplorer.addLocation', () => {
```

Comment:
If you creating suite for serverExplorer.addLocation method, please move all related tests here.